### PR TITLE
Bugfix / change status number for different token errors

### PR DESCRIPTION
**Pull request by adishimoni8**

### CODE REVIEW

File: drfpasswordless/serializers.py [hunk 1]
```diff
@@ -218,40 +218,40 @@ def validate(self, attrs, user=None):
             callback_token = attrs.get('token', None)
             user = user or User.objects.filter(**{alias_attribute_name: alias}).first()
 
-            token = CallbackToken.objects.get(**{'user': user,
+            tokens = CallbackToken.objects.filter(**{'user': user,
```
Comment (yuval-askal) [inline drfpasswordless/serializers.py]: no need to be a dictionary

File: drfpasswordless/serializers.py [hunk 2]
```diff
@@ -218,40 +218,40 @@ def validate(self, attrs, user=None):
...
-                                                 'is_active': True})
+                                                 'to_alias': attrs.get('mobile', None) or attrs.get('email', None),})
 
-            if token.user == user:
-                # Check the token type for our uni-auth method.
-                # authenticates and checks the expiry of the callback token.
-                if not user.is_active:
-                    msg = _('User account is disabled.')
-                    raise serializers.ValidationError(msg)
+            if not tokens.count():
```
Comment (yuval-askal) [inline drfpasswordless/serializers.py]: very rare situation, and users don't have many tokens in the first place so let's minimize db calls and fetch all then filter in Python

File: drfpasswordless/serializers.py [hunk 3]
```diff
@@ -218,40 +218,40 @@ def validate(self, attrs, user=None):
...
+                msg = _(str(InvalidCallbackToken()))
+                raise serializers.ValidationError(msg)
 
-                if api_settings.PASSWORDLESS_USER_MARK_EMAIL_VERIFIED \
-                        or api_settings.PASSWORDLESS_USER_MARK_MOBILE_VERIFIED:
-                    # Mark this alias as verified
-                    user = User.objects.get(pk=token.user.pk)
-                    success = verify_user_alias(user, token)
+            token = tokens.get(**{'is_active': True})
```
Comment (yuval-askal) [inline drfpasswordless/serializers.py]: same comment on kwargs

File: drfpasswordless/serializers.py [hunk 4]
```diff
@@ -218,40 +218,40 @@ def validate(self, attrs, user=None):
...
+                msg = _('User account is disabled.')
+                raise serializers.ValidationError(msg)
 
-                attrs['user'] = user
-                token.delete()
-                return attrs
+            if api_settings.PASSWORDLESS_USER_MARK_EMAIL_VERIFIED \
+                    or api_settings.PASSWORDLESS_USER_MARK_MOBILE_VERIFIED:
+                # Mark this alias as verified
+                user = User.objects.get(pk=token.user.pk)
```
Comment (yuval-askal) [inline drfpasswordless/serializers.py]: equivalent to 

```python
                user = token.user
```

File: drfpasswordless/serializers.py [hunk 5]
```diff
@@ -218,40 +218,40 @@ def validate(self, attrs, user=None):
...
+            return attrs
 
-            else:
-                msg = _('Invalid Token')
-                raise serializers.ValidationError(msg)
         except CallbackToken.DoesNotExist:
-            msg = _(str(InvalidCallbackToken()))
-            raise serializers.ValidationError(msg)
+            raise AuthenticationFailed()
         except User.DoesNotExist:
```
Comment (yuval-askal) [inline drfpasswordless/serializers.py]: if you change 

```python
user = User.objects.get(pk=token.user.pk)
```

then make sure to change catching to the KeyError and check what key was used (or raise above instead)

**Review by yuval-askal — APPROVED** [review]
Looks great!! Just a few minor comments that could optimize DB connections

**Review by BennySaeedian — APPROVED** [review]
GJ

**Review by alexeiMininLanda — APPROVED** [review]

**Review by talcolanda — APPROVED** [review]
looks good :heart:

**Pull request merged**